evennia/utils/evmenu.py

```diff
@@ -1126,17 +1226,174 @@ def node_formatter(self, nodetext, optionstext):
+                    newtext.append(item)
+            else:
+                newtext.append(' ')
+
+        newoptions = []
+        for line in options.split('\n'):
+            if line == '':
+                newoptions.append(' ')
+            else:
+                newoptions.append(line)
+
+        reps = int(.9 + ((len(newtext) + len(newoptions) - 4) / 8))
+        newlength = 5 + reps * 8
+        reassemble = []
+        for line in newtext:
+            reassemble.append(line)
+        for x in range(newlength - (len(newtext) + len(newoptions)) - 1):
+            reassemble.append(' ')
+        for line in newoptions:
+            reassemble.append(line)
+
+        shape = []
+        iwidth = total_width - (margins * 2 + padding + 2)
+        m = ' ' * margins
+        shape.append('  ' + '_' * (iwidth + 3) + '__')
+        shape.append(' /' + m + ANSIljust(reassemble[0],iwidth) + '/ \\ ')
+        shape.append('|' + m + ANSIljust(reassemble[1],iwidth) + '| \\/ ')
+        shape.append('|' + m + ANSIljust(reassemble[2],iwidth) + '| ')
+        shape.append(' \\' + m + ANSIljust(reassemble[3],iwidth) + '\\ ')
+        count = 0
+        for loop in range(reps):
+            shape.append('  \\' + m + ANSIljust(reassemble[4 + count * 8], iwidth) +
+                         '\\ ')
+            shape.append('   |' + m + ANSIljust(reassemble[5 + count * 8], iwidth) +
+                         '| ')
+            shape.append('   |' + m + ANSIljust(reassemble[6 + count * 8], iwidth) +
+                         '| ')
+            shape.append('  /' + m + ANSIljust(reassemble[7 + count * 8], iwidth) +
+                         '/ ')
+            shape.append(' /' + m + ANSIljust(reassemble[8 + count * 8], iwidth) +
+                         '/ ')
+            shape.append('|' + m + ANSIljust(reassemble[9 + count * 8], iwidth) +
+                         '| ')
+            shape.append('|' + m + ANSIljust(reassemble[10 + count * 8], iwidth) +
+                         '| ')
+            shape.append(' \\' + m + ANSIljust(reassemble[11 + count * 8], iwidth) +
+                         '\\ ')
+            count = count + 1
+        shape.append(' _\\' + '_' * (iwidth + 2) + ' \\ ')
+        shape.append('/ ' + ' ' * (iwidth + 2) + '/\\ | ')
+        shape.append('\\' + '_' * (iwidth + 3) + '\\_/ ')
+        final = ''
+        for line in shape:
+            data = line
+            final = final + ' ' * (padding + cpad) + data + '|/'
+        return final
+
+def ANSIljust(input, width):
+    """
+    .ljust doesn't really work on ANSIString, so this is a simple replacement
+
+    input (str): String with ANSI values that needs to be left aligned.
+    width (int): The length input needs to be padded to.
+
+    Returns:
+        result (str): input padded to width
+    """
+
+    result = input
+    t = strip_ansi(input)
+    a = len(t)
+    result = result + ' ' * (width - a)
+    return result
```

Comment:
We shouldn't do workarounds of our own library code. This should be a Ticket against ANSIString and possibly a PR to fix the actual issue.

Reply:
This only was added to support ``scroll`` so we can easily pull it.

Reply:
Actually, as I get ready to submit a ticket it looks like the issue might have been an error in my understanding of ANSIString. .ljust does seem to do just fine when the ANSIString is dealing with color data. The issue is when the string has embeded codes to generate a link in the webclient.

Either way, I've already pulled the function from the since it is no longer needed.